src/DFMM.sol

```diff
@@ -237,13 +237,14 @@ contract DFMM is IDFMM {
 
         if (!state.valid) revert InvalidInvariant(state.invariant);
 
-        _pools[poolId].totalLiquidity += state.deltaLiquidity;
-
         if (_pools[poolId].feeCollector != address(0)) {
-            uint256 fees = state.deltaLiquidity * _pools[poolId].controllerFee
-                / FixedPointMathLib.WAD;
-            if (fees == 0) ++fees;
+            uint256 fees =
+                state.deltaLiquidity.mulWadDown(_pools[poolId].controllerFee);
```

Comment:
Also, should we be checking if controllerFee > 0 in the if?

Reply:
Maybe there's a case where you want to burn liquidity (dumb idea ik), in which case you might want non-zero controllerFee but feeCollector == address(0)?

Reply:
> Maybe there's a case where you want to burn liquidity (dumb idea ik), in which case you might want non-zero controllerFee but feeCollector == address(0)?

Yeah, I think it makes sense to check the controller fee in the if condition instead of the address.

Reply:
If we want to prevent swap from burning liquidity we can still reintroduce the address check later.